docs/namespaced-broker-resources.md

```diff
@@ -0,0 +1,159 @@
+
+### Limiting Access to Plans
+
+There are often situations when not all services and plans should be available to all users. A cluster administrator may wish to only provide free plans to certain users or restrict the ability to provision very expensive services. Additionally, when developers are creating new services that are exposed by brokers, they want to be able to iterate on those services without exposing them to all users in a cluster.
+
+Service Catalog's cluster-scoped resources for brokers, services, and plans are not sufficient to implement access control to ensure that users have access only to the service and plans that they should. For these resources, application of RBAC is really centered around what is visible to them, but is not enforced when a provision request is issued. For example, a `CLusterRole` could be created to prohibit a given user or group to view classes and plans, but this can not be used at provisioning time to ensure that the user cannot create the resource. Namespace-scoped brokers, services and plans, however, can be effectively combined with Kubernetes role based access control and Service Catalog Catalog Restrictions in order to provide more granular control over service instance provisioning.
+
+## Enabling Namespace Scoped Broker Resources
+
+Currently, namespace-scoped broker resources are an alpha-feature of Service Catalog behind a feature flag. To start using these resources, you will need to pass an argument to the API Server when you install Service Catalog: `--feature-gates NamespacedServiceBroker=true`.
+
+If you are using Helm, this flag has been surfaced as a chart parameter:
+
+```console
+helm install svc-cat/catalog \
+   --name catalog \
+   --namespace catalog \
+   --set namespacedServiceBrokerEnabled=true
+```
+
+If you are also using RBAC, providing this option to the Helm install command will also create the needed RBAC resources.
+
+## Using Namespace Scoped Broker Resources
+
+Once Service Catalog has been installed with this feature gate enabled, you should see three new resource types: `ServiceBroker`, `ServiceClass`, and `ServicePlan`.
+
+In order to register a `ServiceBroker` resource, you create a YAML definition that looks similar to a `ClusterServiceBroker`. This resource will use a different resource kind, however, and requires a namespace. An example might look like:
+
+```yaml
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ServiceBroker
+metadata:
+  name: example-ns-broker
+  namespace: ns-broker
+spec:
+  authInfo:
+    basic:
+      secretRef:
+        name: my-service-broker-auth
+        namespace: broker
+  url: http://my-service-broker.broker.svc.cluster.local
+```
+
+Once this resource is created, Service Catalog will query the Service Broker for the list of available Services and create corresponding `ServiceClass` and `ServicePlan` resources. These resources might look like this:
+
+```yaml
+apiVersion: servicecatalog.k8s.io/v1beta1
```

Comment:
suggestion: You may want to trim down fields from the yaml that aren't necessary to demonstrate what a namespaced class and plan look like. Just so that it's not an intimidating wall of text. Since they aren't copy/pasting the yaml, it's not critical that it's complete.

e.g.

```
apiVersion: servicecatalog.k8s.io/v1beta1
kind: ServiceClass
metadata:
  name: 25434f16-d762-41c7-bbdd-8045d7f74ca6
  namespace: ns-broker
  resourceVersion: "13"
spec:
  bindable: true
  bindingRetrievable: false
  description: MySQL
  externalID: 25434f16-d762-41c7-bbdd-8045d7f74ca6
  externalName: mysql-5-7
  planUpdatable: false
  serviceBrokerName: example-ns-broker
```